plugins/callback/opentelemetry.py

```diff
@@ -288,6 +291,60 @@ def set_span_attribute(self, span, attributeName, attributeValue):
             if attributeValue is not None:
                 span.set_attribute(attributeName, attributeValue)
 
+    def add_network_attributes_if_possible(self, task_data, span):
+        """ update the span attributes with the service that the task interacted with, if possible """
+
+        if any(s in task_data.action for s in ('get_url', 'homebrew_tap', 'uri', 'win_get_url', 'win_uri')) and task_data.args.get('url', None):
+            self.add_network_attributes(task_data.args.get('url'), span)
+
+        # Support apt_repository: repo: deb https://artifacts./...packages/6.x/apt stable main
+
+        if 'win_chocolatey' in task_data.action:
+            if task_data.args.get('source', None):
+                self.add_network_attributes(task_data.args.get('source'), span)
+            elif task_data.args.get('proxy_url', None):
+                self.add_network_attributes(task_data.args.get('proxy_url'), span)
+            else:
+                self.set_span_attribute(span, "net.peer.name", 'chocolatey')
+                self.set_span_attribute(span, "rpc.service", 'chocolatey')
+
+        if 'homebrew' in task_data.action:
+            self.set_span_attribute(span, "net.peer.name", 'homebrew')
+            self.set_span_attribute(span, "rpc.service", 'homebrew')
+
+        if 'git' in task_data.action:
+            self.add_network_attributes(task_data.args.get('repo'), span)
+
+        if 'yum' in task_data.action and task_data.args.get('name', '').startswith('http'):
+            self.add_network_attributes(task_data.args.get('name'), span)
+
+        if 'yum_repository' in task_data.action:
+            if task_data.args.get('baseurl', None):
+                self.add_network_attributes(task_data.args.get('baseurl'), span)
+            elif task_data.args.get('mirrorlist', None):
+                self.add_network_attributes(task_data.args.get('mirrorlist'), span)
+            else:
+                self._display.debug('yum_repository without a baseurl or mirrorlist.')
+
+        if 'zypper_repository' in task_data.action and task_data.args.get('repo', None):
+            self.add_network_attributes(task_data.args.get('repo'), span)
```

Comment:
this section is probably something that will grow and grow, the logic is more or less the same, maybe it is better to transform the input in data

```
addActionAttributes('homebrew', [{'name': 'net.peer.name', 'value':'homebrew' }, {'name': 'rpc.service', 'value':'homebrew' }])
addNetworkAttributes('yum_repository', [{'name': 'baseurl' }, {'name': 'mirrorlist'}])

def addActionAttributes(action, attributes):
    if task_data.action == action 
        for attribute in attributes:
            if task_data.args.get(attribute['name'], None):
                self.set_span_attribute(span, attribute['name'], attribute['value'])

def addNetworkAttributes(action, attributes):
    if task_data.action in action 
        for attribute in attributes:
            if task_data.args.get(attribute['name'], None):
                self.add_network_attributes(span, attribute['name'], span)
```